### PR TITLE
feat(c-api): advanced parameter API + streaming surface (compressStream2 / decompressStream)

### PR DESCRIPTION
**Pull request by polaz**
## Summary

Phase 6.2 slice 1 of #127: the stable advanced-parameter and streaming C API surface. The advanced dictionary attach surface (`ZSTD_CCtx_loadDictionary` / `refCDict` / `refPrefix` families) and the `ZSTD_estimate*` static-API queries follow in a second PR — the combined diff would cross the review-size threshold (~1.7k lines here already).

## Changes

**Codec (`zstd`):**
- `FrameCompressor::set_content_size_flag` / `set_dictionary_id_flag` (upstream `ZSTD_c_contentSizeFlag` / `ZSTD_c_dictIDFlag` semantics, both default on). Flag off omits the FCS field (forcing the windowed layout) or the dictionary ID; round-trip tests for both, ID-less dict frames decode via the explicit-dictionary path.

**C ABI (`c-api`), 25 new symbols:**
- **Parameters**: `ZSTD_cParam_getBounds` / `ZSTD_dParam_getBounds` (`ZSTD_bounds`, 16-byte ABI snapshot), `ZSTD_CCtx_setParameter` / `getParameter` over the full v1.5.7 stable `ZSTD_cParameter` set (sticky `CCtxParams`; 0 = auto; workers accepted but single-threaded per the issue's 6.4 deferral), `ZSTD_DCtx_setParameter` / `getParameter` (`windowLogMax`), `ZSTD_CCtx_setPledgedSrcSize`, `ZSTD_CCtx_reset` / `ZSTD_DCtx_reset` (session / parameters / both).
- **`ZSTD_compress2`**: one-shot over the sticky set, pledge validated against srcSize (`srcSize_wrong`), checksum / content-size / dict-id flags land in the frame header.
- **Streaming**: `ZSTD_inBuffer` / `ZSTD_outBuffer` (24-byte ABI snapshots), `ZSTD_createCStream` / `freeCStream` / `createDStream` / `freeDStream` (context aliases), `ZSTD_compressStream2` (continue / flush / end over a push-side `StreamingEncoder` with a pending-drain buffer), legacy `initCStream` / `compressStream` / `flushStream` / `endStream` / `initDStream`, `ZSTD_decompressStream` over `decode_from_to` with explicit frame restarts at frame boundaries, `windowLogMax` enforcement from the parsed header, and the In/Out size hints.

Notable: a reused decoder parks on its finished frame, so without the explicit per-frame restart a multi-frame stream made no forward progress and a spec-conformant caller loop hung — caught by the two-frames-back-to-back test.

## Review-round additions

- **`ZSTD_c_targetCBlockSize` is applied** (was accepted-but-inert): `FrameCompressor` / `StreamingEncoder` gained `set_target_block_size`, capping every physical block's payload at the target (raw length ≤ target bounds the compressed length, so wire blocks land at or under target + 3 header bytes — a strict form of upstream's best-effort convergence); plumbed into `ZSTD_compress2` and the streaming frame start; `MIN_TARGET_BLOCK_SIZE` names the 1340 floor.
- **Stage guards**: `ZSTD_compress2` returns `stage_wrong` while a streaming frame is in flight; `ZSTD_DCtx_setParameter` and the parameters-only `ZSTD_DCtx_reset` reject mid-frame calls.
- **`ZSTD_decompressStream` header-error classification**: truncated header keeps the need-more-input hint, malformed header surfaces the decode error (the hint with nothing consumed used to spin callers), fully-buffered skippable frames are consumed whole.
- **`D_WINDOW_LOG_MAX`** is pointer-width dependent (30 on 32-bit / 31 on 64-bit, matching the vendored header).
- **Advisory-hint reserve cap**: the owned block loop trusts a streaming size hint only up to a 64 KiB lookahead past bytes actually read (an exact one-shot length keeps the precise reservation).
- **v0.0.35 groundwork**: version bump + changelog, standing in for the release automation's release PR (its baseline diff packages the v0.0.34 tag worktree, which fails on that tree's bare path dependency; tagging v0.0.35 on a tree with the fixed manifest gives it a packageable baseline).
- Weekly npm downloads badge in the README.

## Testing

- 880 workspace tests (9 new c-api Rust tests: ABI size snapshots, bounds coverage, set/get/reset, compress2 round-trip + pledge mismatch, chunked 1 MiB streaming round-trip through 1 KiB output buffers, flush + two frames on one context, window-limit rejection, size hints)
- `c_consumer.c` extended: params section + chunked streaming round-trip through the vendored header and the drop-in link name; compiled, linked, and run green locally
- Clippy clean (workspace, `-D warnings`, both feature combos); `thumbv6m-none-eabi` check clean; doc tests green

Part of #127

## Summary by CodeRabbit

* **New Features**
  * Advanced sticky parameter controls for compressor/decompressor contexts
  * Streaming compression/decompression API with buffer helpers and legacy wrappers
  * One-shot compression that uses context-stored parameters
  * Options to omit frame content-size and dictionary-ID from headers
  * Configurable target block size for emitted blocks

* **Tests**
  * Expanded C-ABI and streaming round-trip tests covering parameter bounds, resets, pledged-size checks, block sizing, and edge cases

### CODE REVIEW

File: c-api/src/params.rs [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+            // it as auto like upstream's special-value rule.
+            p.target_length = opt(value);
+        }
+        ZSTD_C_STRATEGY => {
+            p.strategy = if value == 0 {
+                None
+            } else {
+                Strategy::from_ordinal(value as u32)
+            };
+        }
```
[the remark this answers is not in the document]
Comment (polaz) [reply c-api/src/params.rs:342]: @coderabbitai Fixed — the strategy arm now returns parameter_outOfBound when from_ordinal returns None for a non-zero value.

File: c-api/src/streaming.rs [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn ZSTD_initDStream(zds: *mut ZSTD_DCtx) -> usize {
+    if zds.is_null() {
+        return encode(ZSTD_ErrorCode::ZSTD_error_GENERIC);
+    }
+    let dctx = unsafe { &mut *zds };
+    dctx.reset_session();
+    dctx.stream_frame_done = true;
+    ZSTD_DStreamInSize()
+}
```
[the remark this answers is not in the document]
Comment (polaz) [reply c-api/src/streaming.rs:406]: @coderabbitai Fixed — removed the redundant assignment; reset_session owns the flag.

File: c-api/src/tests.rs [hunk 3]
```diff
@@ -556,3 +556,400 @@ fn create_cdict_rejects_non_dictionary() {
...
+};
+use core::ffi::c_int;
+
+// ABI invariants: struct sizes match upstream `sizeof` on 64-bit.
+#[test]
+fn streaming_buffer_abi_sizes() {
+    assert_eq!(core::mem::size_of::<ZSTD_inBuffer>(), 24);
+    assert_eq!(core::mem::size_of::<ZSTD_outBuffer>(), 24);
+    assert_eq!(core::mem::size_of::<ZSTD_bounds>(), 16);
+}
```
[the remark this answers is not in the document]
Comment (polaz) [reply c-api/src/tests.rs:581]: @coderabbitai Fixed — the ABI size test is now gated on target_pointer_width = "64", mirroring frame_header_abi_layout_is_stable.
